### PR TITLE
[Bug Fix] Invite link Redirect

### PR DESCRIPTION
**Pull request by d3li0n**
# Description

Fixes an issue where users were not redirected to invite link page when they created or signed in to an account

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [x] Added unit tests
- [x] Manually verified each sign and registration endpoint to verify the bug is fixed

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/api-client/index.ts [hunk 1]
```diff
@@ -450,12 +445,6 @@ class APIClient {
     toggleOff: async (insightName: string): Promise<void> =>
       this.req('DELETE', `/api/v1/insights`, undefined, { insightName }),
   }
-  image = {
```
Comment (d3li0n) [inline packages/api-client/index.ts:453]: Controller no longer exist

File: packages/app/pages/course/[cid]/invite.tsx [hunk 2]
```diff
@@ -21,10 +21,6 @@ export default function Invite(): ReactElement {
   const cardMetaDescription = `This course is managed by ${course?.organizationCourse.name}`
 
   useEffect(() => {
-    if (!profile) {
-      localStorage.setItem('lastVisited', window.location.href)
```
Comment (d3li0n) [inline packages/app/pages/course/[cid]/invite.tsx:25]: We had bunch of issues with this code, so I moved it to be backend and store it as cookies

File: packages/server/src/course/course.controller.ts [hunk 3]
```diff
@@ -210,6 +208,15 @@ export class CourseController {
       courseInviteCode: courseWithOrganization.courseInviteCode,
     };
 
+    res.cookie(
+      '__SECURE_REDIRECT',
+      Buffer.from(`/course/${id}/invite?code=${code}`).toString('base64'),
```
Comment (d3li0n) [inline packages/server/src/course/course.controller.ts:213]: Base64 is better to store this information as `/` is modified to `%2F`

**Comment by AdamFipke** [comment]
I have found a bug while trying to manually test this.
Steps to reproduce:
1. Create invite code (I did mine a bob) and copy the link
2. Paste the link into another browser that is not logged into the system
3. You get redirected to the login page (as expected). Login.
4. Get stuck on infinite loading screen. I noticed the URL is http://localhost:3000/course/2/invite?code=bob%0D%EF%BF%BD%EF%BF%BD and when I try deleting all of the %XX stuff off the end of it (so that it's just code=bob), it works just fine

**Comment by AdamFipke** [comment]
> I have found a bug while trying to manually test this. 

Nevermind, seems like the bug was a one time-thing. I could replicate it initially with the first course I tried, but couldn't replicated the bug for future courses I tried. A strange mystery

**Review by AdamFipke — APPROVED** [review]
Nice, seems to work pretty good

**Pull request merged**